prime-router/src/main/kotlin/fhirengine/engine/FHIRConverter.kt

```diff
@@ -98,124 +100,147 @@ class FHIRConverter(
-                    val bodyBytes = FhirTranscoder.encode(bundle).toByteArray()
-                    val blobInfo = BlobAccess.uploadBody(
-                        Report.Format.FHIR,
-                        bodyBytes,
-                        report.name,
-                        queueMessage.blobSubFolderName,
-                        routeEvent.eventAction
-                    )
+                        // upload to blobstore
+                        val bodyBytes = FhirTranscoder.encode(bundle).toByteArray()
+                        val blobInfo = BlobAccess.uploadBody(
+                            Report.Format.FHIR,
+                            bodyBytes,
+                            report.name,
+                            queueMessage.blobSubFolderName,
+                            routeEvent.eventAction
+                        )
 
-                    // track created report
-                    actionHistory.trackCreatedReport(routeEvent, report, blobInfo = blobInfo)
-                    azureEventService.trackEvent(
-                        ReportCreatedEvent(
-                            report.id,
-                            queueMessage.topic
+                        // track created report
+                        actionHistory.trackCreatedReport(routeEvent, report, blobInfo = blobInfo)
+                        azureEventService.trackEvent(
+                            ReportCreatedEvent(
+                                report.id,
+                                queueMessage.topic
+                            )
                         )
-                    )
 
-                    FHIREngineRunResult(
-                        routeEvent,
-                        report,
-                        blobInfo.blobUrl,
-                        FhirRouteQueueMessage(
-                            report.id,
+                        FHIREngineRunResult(
+                            routeEvent,
+                            report,
                             blobInfo.blobUrl,
-                            BlobAccess.digestToString(blobInfo.digest),
-                            queueMessage.blobSubFolderName,
-                            queueMessage.topic
+                            FhirRouteQueueMessage(
+                                report.id,
+                                blobInfo.blobUrl,
+                                BlobAccess.digestToString(blobInfo.digest),
+                                queueMessage.blobSubFolderName,
+                                queueMessage.topic
+                            )
                         )
+                    }.collect(Collectors.toList())
+                }
+            } else {
+                val nextEvent = ProcessEvent(
+                    Event.EventAction.NONE,
+                    queueMessage.reportId,
+                    Options.None,
+                    emptyMap(),
+                    emptyList()
+                )
+                val report = Report(
```

Comment:
So this really is an empty report...none of the bundles make it through so it should have zero items.  However, due to some checks in the lineage generation, that code will actually throw an exception if you create a report without any items/and item lineage.

There is tech there to clean for sure, but I'm going to add a ticket and defer that.